### ISSUE TITLE
Trader Expansion Feature Request 2.0

### ISSUE THREAD

**Issue by kilozombie** [d3athrow/vgstation13#13364]
After a small bit of discussion in a google doc, I've organized a list of ways to improve Trader and give the job more things to do. The entire idea is to add a way for them to get rare or interesting items to sell that are
1. useful and/or fun for the crew
2. not very useful for the traders

Hopefully, if everything goes well, perhaps tradertiding will die down. This phenomenon occurs, I think, largely because the job has very little to do, and people think you've gotta steal shit to be interesting. I mean, maybe. But it's fucking annoying and instead of just banning everyone who does it, maybe this is a slightly smarter idea.

Everything below is just a loose mishmash of ideas from the document.

Proposed systems of delivering trader items:
1. Vending machine.
    Vending machines are the basis of this idea, and while they might work for one category at once, it's hard to give traders multiple tiers of items. For instance, how do you give traders that rare bottle of peridaxon, but also some cool clothes and toys to sell?
2. Crates with random contents.
    Crates, with proper balancing and use of a weighting system, could produce a high variety of items for a trader. However, it’s difficult to give traders any choice in this. They’re at the will of the RNG for what items they get. Also, completely random people could come and loot the outpost for the items, making the job completely pointless.
3. Cargo shuttle with special Trader-specific contents.
    A shuttle could have costs for specific items, allowing coders to balance certain items over other ones. For instance, an expensive bottle of rare xenobiology drugs would cost more than a few cosmetic costumes, but **it's random what selection of purchasable goods are in the shuttle**. This and an uplink could have a steady stream of income as well as new random goods, keeping rounds interesting even if the Trader manages to sell their initial stock. It turns Trader into more of a fleshed out job, rather than just a vague, nebulous concept of ‘trading what’s on your outpost lol’.
4. Uplink or teleporter relay.
    An uplink could have costs for specific items, allowing coders to balance certain items over other ones. For instance, an expensive bottle of rare xenobiology drugs would cost more than a few cosmetic costumes, but **it's random what selection of purchasable goods are in the uplink**. See cargo for advantages of a rolling stock. If this was stationary on the outpost, like a teleporter relay, it also prevents Traders running around buying new things, and encourages them to go for supply runs back and forth. A relay is my personal favorite choice-- I think it does everything well to keep the job interesting without introducing any way to powergame in particular.
    
The idea with items are to make them nonviable for a Trader to use, but valuable to a Crew. This encourages a relationship of commerce without allowing the birds to powergame with their items, even as an antag. This means weapons and strong toys should generally be out of the list.
    
Potential item list (not final, please suggest and critique specific items):
- Box of marauder circuits (1 missing)
- Box of natural bluespace crystals
- Flawless bluespace crystal
- Bluespace beakers
- Advanced magboots
    Vox cannot wear these, and have their own even better pair at roundstart.
- Bottle of peridaxon
- Bottle of nanites
- Bottle of rezadone
- Random virus sample dish
    Vox doesn't have Virology equipment and can't use this to cause havoc.
- Borer egg
    Technically these can already be obtained with accurate no-fruit pie shenanigans.
- Clown spacesuit and helmet.
- Small number of Phazon sheets.
- Bananium sheets.
- Single bananium coin.
- Paper with location and identification of two Genetics powers.
    Two or one? Shouldn't be an injector, else Traders will just put good powers in themselves.
- Some sort of unanchored machinery?
- Random clothing/costume crate
    The more variety, the better. In an uplink/teleporter or cargo shuttle implementation, these should be cheap, so you can get a lot of them if that's your thing and they happen to show up.
- Box of random advanced machinery parts
    R&D hardly makes these for other departments. If traders could offer up a kit to DIY upgrade your workplace, that'd be pretty cool.
- Unobtainable food items, or food with an unobtainable modifier.
    Golden food? Magic food? Bluespace sweet sundae ramen? Nar-Sliders? Who knows, it's completely up to the coders. There aren’t currently any adminbus food items that I know of.
    Possibly some special Vox crackers? Feed to Poly for special effects??
- Lazarus injectors.
    Miners rarely ever give these out, and every department excluding Cargo has pets that they don't want to die. Sell the HoP an injector for Ian!
- Cages with random animals in them.
    Specifically slimes? Or just all kinds of mobs? Is this a golden-slime situation where it's usually not worthwhile to open?
- Triple capacity fire extinguisher.
    For ease of Xenobiology. Also potentially great with dangerous chemicals. Need input on if this can be exploited badly.
- Xenoarch-esque modified items.
    I don’t know all the details of Xenoarch. Do small artifacts (wrenches made of aluminum or whatever) have any effects? Less, more brute damage? Could we get Vox traders selling some tacky adamantine clown shoes for an insane price?
- Plastic, cardboard, wired tiles, carpet, etc.
    Some of this spawns in moderate quantities at random on Box and Meta. They’re not obtainable from mining IIRC but a nice sellable item.
- Portable recharger.
    Presumably would recharge at ⅓ or ¼ the normal rate, but extremely handy if you’re a Security officer always on the run. Stick it in your backpack and juggle your energy weapons around.
- A variety of implants.
    Potentially fun to sell. Non-crew birds wouldn’t have things like loyalty implants, though.
- Special or improved borg parts/modules.
    Vox can’t use these for themselves, as there are few lawsets that respect non-crew non-humans, and they don’t exactly have a Robotics bay. Also, borgs will now want to buy parts from traders. Wew.
- Infinite ammo foam dart pistol, cap gun, laser tag gun, etc.

Hopefully the end result is that normal goods that people already make (botany on box, chems and drinks and weapons on deff, misery on metaclub) aren't invalidated, but people who don't feel like doing those can still have plenty of fun and make the round more interesting.

Yes, that's right. This will make the round more interesting for both traders and crew. Now instead of buying some obscure food items or rarely something slightly useful, you can get something specific to your job that changes up the game a little bit, but only if you INTERACT with TRADERS.

So, y'know, that whole "make departments" interact meme that so many people are a fan of.

anyway, TL;DR it's wyci, so hopefully somebody comes by who's interested in coding it.

**Comment by Shadowmech88** [comment]
I saw an idea in the thread a couple weeks ago that I thought was neat.

Give traders a way to spend a bunch of money in order to stash items from the current round, to be retrieved some other round.

**Comment by kilozombie** [comment]
I fucking loved that idea, but it seems like it'd be a pain to code. If it's not I'd be willing to write something simple up for it though.
If I recall, the idea was basically just: there's a safe for one ckey attached to one staticname (or not?), and you can put items in and retrieve them later. With a hefty blacklist, so no armor, guns, traitor items, etc.
And I'd also like some way to turn in items you barter for for some end-round cash, like a Vox Sale Relay. Sell the emags and viscerator nades instead of storing them!

**Comment by Probe1** [comment]
how would we even begin to code and balance that

**Comment by Shadowmech88** [comment]
>how to code

DM supports reading from and writing to text files.
We could store a list of all stored items (as well as associated ckeys, if desired) in a text file and then read from it to get the items out.
This would be pretty easy as long as we only needed to withdraw a new item of the same type, rather than that *specific* item with all its vars at the time intact.

>how to balance

A thoroughly huge blacklist of items deemed too powerful to allow to be taken between rounds, and/or preventing people from choosing which item they get and making it a pot luck.

**Comment by SysadminJeroen** [comment]
>Triple capacity fire extinguisher.
We already have a lot of water in it, and it fits on your back.

And if you're planning to save the stuff in a text file, I suggest just
saving it like anything else that is saved, like preferences.

You should only be able to store money that way, after each round end
between 5 and 15% of the money should disappear. If there is less than 100
credits in the vault for the ckey the account should be deleted from the
vault.
The money disappearing after each round end should encourage Vox to
frequently play vox.

Op 10 jan. 2017 06:13 schreef "Shadowmech88" <notifications@github.com>:

how to code

DM supports reading from and writing to text files.
We could store a list of all stored items (as well as associated ckeys, if
desired) in a text file and then read from it to get the items out.

how to balance

A thoroughly huge blacklist of items deemed too powerful to allow to be
taken between rounds, and/or preventing people from choosing which item
they get and making it a pot luck.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/d3athrow/vgstation13/issues/13364#issuecomment-271488376>,
or mute the thread

**Comment by SysadminJeroen** [comment]
I should add that if a player has selected that they want to become a vox,
and if they did not become one because it is full then they should only
lose 1% of their money.

Op 10 jan. 2017 06:22 schreef "Jeroen de Neef" <jeroen52@gmail.com>:

>Triple capacity fire extinguisher.
We already have a lot of water in it, and it fits on your back.

And if you're planning to save the stuff in a text file, I suggest just
saving it like anything else that is saved, like preferences.

You should only be able to store money that way, after each round end
between 5 and 15% of the money should disappear. If there is less than 100
credits in the vault for the ckey the account should be deleted from the
vault.
The money disappearing after each round end should encourage Vox to
frequently play vox.

Op 10 jan. 2017 06:13 schreef "Shadowmech88" <notifications@github.com>:

how to code

DM supports reading from and writing to text files.
We could store a list of all stored items (as well as associated ckeys, if
desired) in a text file and then read from it to get the items out.

how to balance

A thoroughly huge blacklist of items deemed too powerful to allow to be
taken between rounds, and/or preventing people from choosing which item
they get and making it a pot luck.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/d3athrow/vgstation13/issues/13364#issuecomment-271488376>,
or mute the thread

**Comment by kilozombie** [comment]
What about a way to sell certain items that you barter for, as well as the bank account?
I'm hesitantly liking the idea that you slowly lose money from the account over time. I'd play way more often if that was the case, though I could see it encouraging traders to raid the vault or something else ridiculous.

**Comment by Kurfursten** [comment]
>10% rate of decay, but stay above 100 credits
>Average rounds per day: about 16 (sample size 5 days)
>How many rounds 1 million credits would last: 87 (~5.5 days)
>Minimum credits to keep your vault open for one day of rest: ~550 (of course, this would be very risky because if there are more than sixteen rounds, you lose it all)
>Average starting money for Vox: ~416 (this is highly random, player has just as much chance to get 2 credits as 2000 credits)

It does seem to encourage metagaming with the miners, or at least, to focus on only selling to them. You trade with a regular greytider and the best you can hope for is a couple hundred which buys you very little time. But make a cool million off one trade with a miner, and you're set for 87 rounds. These kinds of trades would be totally necessary for anyone who has a job. Since the goal of the player is no longer on a single round but building up a fortune to secure their future, there is a very strong incentive to abuse the system. In fact, since miners often set their ATM security to minimum and wander alone on the asteroid, it does seem the best strategy to maintain your vault is to waylay a miner and steal his ID. On the other side of the meta perspective, since players know that killing a Vox before they can stash any earnings for the round will hurt them in a *permanent*, persistent sort of way, I foresee more antagonists going out of their way to rush to the outpost to kill players they dislike. Instead of deporting birds back to the asteroid, I predict security officers will instead opt to throw them in perma to prevent them from stashing. Trader antagonists will naturally have a massive advantage if they have stockpiled a good vault, potentially starting a round immediately with any nook op gear they may have looted in the past, for example. Since there's no reason to "waste" items by ending a shift by not stashing them with a Vox trader, it may become standard operating procedure to quickly drop off all the head of staff gear with a trader metafriend so they can be retrieved in a future round. I would expect that a consistent trader should never go without a ton of hyposprays, in case of emergency.

**Wow, that turned out to be a wall of text.** Short answer, I don't really like the implications invited by persistent items. Call me no-fun if you want, but these are things that need to be talked about at the very least.

**Comment by kilozombie** [comment]
Lot of really great arguments in there, kurf. I didn't include it in the original post because, yeah, even I couldn't think of all those issues. You would have to make a lot of tweaks to get it workable.

**Comment by AI-Is-Rogue** [comment]
>Advanced magboots
Vox cannot wear these, and have their own even better pair at roundstart.

They are able to wear the advanced magboots.

**Comment by Durgen** [comment]
I'm not really a fan of the persistent account idea. It's a huge game changer and doesn't seem necessary for fixing trader. It would surely have huge unintended consequences in the form of different player behavior like Kurfurst mentioned. A powergame meta, and maybe a counter-powergaming meta would emerge and it would probably involve bullying traders.

Here is my ideal solution. It combines a couple of things from Kilozombie's OP.

1. Traders start with 1 or 2 special Trader tokens
2. Tokens can be used to buy crates with RNG trade contents (crates arrive at the outpost on a shuttle, or via Special Vox Bluespace technology)
3. More tokens can be obtained from a special vending on the outpost using credits

**Issues with this and other solutions:**

Letting traders redeem cash for goods won't currently encourage the type of behavior we are trying to encourage. It would reward making friends with miners, and getting a Head/QM to give you their account information, as this is 10-100x more profitable than actually selling things in arrivals. 
For example: last night myself and another trader made 200 credits from hawking improvised weapons and sec gear in arrivals, and 5800 credits from selling the pirate shuttle to a bored QM.

If we are actually going to give credits real value (which this proposed trader fix would definitely do) the Head of Staff accounts and mining dosh need to be rebalanced as well.

**Comment by kilozombie** [comment]
Cash for goods, no. But if we're having money carry over somehow (and it's still pretty much just for fun, so any powergaming is moot), then being able to SELL things would be fine. For instance, sell an emag for 1000 voxbucks that go into your account, perhaps for use in another round.
I don't think credits should have much real value. For traders, it should just be a way for them to look back on their achievements fondly. An encouragement, not a direct reward.

**Comment by Durgen** [comment]
So you are envisioning a persistent leaderboard? I don't really like the sound of that.

Also it would not addresses the primary issue, which is that traders don't have much to trade.

**Comment by kilozombie** [comment]
That's the point of the OP, the persistent account idea is just something somebody brought up.

**Comment by Intigracy** [comment]
`Infinite ammo foam dart pistol`

👍

**Comment by kilozombie** [comment]
Bait security for an eternity
